packages/common/src/content/compose.ts

```diff
@@ -54,6 +54,18 @@ const shouldUseLayerInfo = (
   );
 };
 
+// this logic adapted from hub-indexer's determineName(), see
+// https://github.com/ArcGIS/hub-indexer/blob/8f4dd6f928124c1f35dd02bc11bd996191ee1160/packages/duke/compose/common.js#L7-L34
+const getContentName = (
+  item: IItem,
+  layer?: Partial<ILayerDefinition>,
+  _shouldUseLayerInfo = false
+) => {
+  return (
+    (_shouldUseLayerInfo ? layer.name : item.title || item.name) || ""
```

Comment:
The only thing I worry about with this is that if `_shouldUseLayerInfo` is true and `layer.name` doesn't exist, we'll only get an empty string. Is it possible for a layer to NOT have a name assigned?